File: src/main/java/jenkem/shared/LineWidth.java
package jenkem.shared;

/**
 * Enum for different line Widths.
 */
public enum LineWidth {
    GigaHuge(80), MegaHuge(76), Huge(72), Default(68), Large(64), Medium(56), Small(48), Mini(40), Icon(32), Tiny(16);

    private int value;

    private LineWidth(final int value) {
        this.value = value;
    }

    public String getValueString() {
        return String.valueOf(value);
    }

    public int getValue() {
        return value;
    }

    @Override
    public String toString() {
        return name();
    }
}

File: src/main/scala/jenkem/engine/LineWidth.scala
package jenkem.engine

/**
 * Number of characters in one line of output.
 * May be 1 less depending on X Kick.
 */
object LineWidth {
  sealed abstract class Value(val value: Int)
  case object Huge extends Value(72)
  case object Default extends Value(68)
  case object Large extends Value(64)
  case object Medium extends Value(56)
  case object Small extends Value(48)
  case object Mini extends Value(40)
  case object Icon extends Value(32)
  case object Tiny extends Value(16)

  def default = Default
  def getAll = List(Huge, Default, Large, Medium, Small, Mini, Icon, Tiny)
}
